src/hough.rs

```diff
@@ -248,6 +257,84 @@ mod test {
         assert_eq!(line.angle_in_degrees, 90);
     }
 
+    #[test]
+    fn draw_polar_line_horizontal() {
+        let mut image = GrayImage::new(5, 5);
+        draw_polar_line(&mut image, PolarLine { r: 2f32, angle_in_degrees: 90}, Luma([1u8]));
+        let expected = gray_image!(
+            0, 0, 0, 0, 0;
+            0, 0, 0, 0, 0;
+            1, 1, 1, 1, 1;
+            0, 0, 0, 0, 0;
+            0, 0, 0, 0, 0);
+        assert_pixels_eq!(image, expected);
+    }
+
+    #[test]
+    fn draw_polar_line_vertical() {
+        let mut image = GrayImage::new(5, 5);
+        draw_polar_line(&mut image, PolarLine { r: 2f32, angle_in_degrees: 0}, Luma([1u8]));
+        let expected = gray_image!(
+            0, 0, 1, 0, 0;
+            0, 0, 1, 0, 0;
+            0, 0, 1, 0, 0;
+            0, 0, 1, 0, 0;
+            0, 0, 1, 0, 0);
+        assert_pixels_eq!(image, expected);
+    }
+
+    #[test]
+    fn draw_polar_line_bottom_left_to_top_right() {
+        let mut image = GrayImage::new(5, 5);
+        draw_polar_line(&mut image, PolarLine { r: 3.0, angle_in_degrees: 45}, Luma([1u8]));
+        let expected = gray_image!(
+            0, 0, 0, 0, 1;
+            0, 0, 0, 1, 0;
+            0, 0, 1, 0, 0;
+            0, 1, 0, 0, 0;
+            1, 0, 0, 0, 0);
+        assert_pixels_eq!(image, expected);
+    }
+
+    #[test]
+    fn draw_polar_line_top_left_to_bottom_right() {
+        let mut image = GrayImage::new(5, 5);
+        draw_polar_line(&mut image, PolarLine { r: 0.0, angle_in_degrees: 135}, Luma([1u8]));
+        let expected = gray_image!(
+            1, 0, 0, 0, 0;
+            0, 1, 0, 0, 0;
+            0, 0, 1, 0, 0;
+            0, 0, 0, 1, 0;
+            0, 0, 0, 0, 1);
+        assert_pixels_eq!(image, expected);
+    }
+
+    macro_rules! test_detect_line {
+        ($name:ident, $r:expr, $angle:expr) => {
+            #[test]
+            fn $name() {
+                let options = LineDetectionOptions {
+                    vote_threshold: 10,
+                    suppression_radius: 8,
+                };
+                let mut image = GrayImage::new(100, 100);
+                draw_polar_line(&mut image, PolarLine { r: $r, angle_in_degrees: $angle}, Luma([255u8]));
+
+                let detected = detect_lines(&image, options);
+                assert_eq!(detected.len(), 1);
+
+                let line = detected[0];
+                assert_approx_eq!(line.r, $r, 1.1);
+                assert_approx_eq!(line.angle_in_degrees as f32, $angle as f32, 5.0);
+            }
+        };
+    }
+
+    test_detect_line!(detect_line_50_45, 50.0, 45);
+    test_detect_line!(detect_line_eps_135, 0.001, 135);
+    // https://github.com/PistonDevelopers/imageproc/issues/280
+    test_detect_line!(detect_line_neg10_120, -10.0, 120);
```

Comment:
Those look like a good fit for a benchmark test as well, finishing the TODO below.